src/main/java/org/broadinstitute/dsm/model/gbf/GBFOrderGateKeeper.java

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.dsm.model.gbf;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Uses a {@link GBFOrderTransmitter} and {@link GBFOrderFinder}
+ * to find and transmit orders to GBF for a given
+ * DDP instance
+ */
+public class GBFOrderGateKeeper {
+
+    private final GBFOrderFinder orderFinder;
+    private final GBFOrderTransmitter transmitter;
+    private final String ddpInstanceName;
+    private static final String UPDATE_TRANSMISSION_DATE = "update ddp_kit_request req set req.transmitted_at = ? where external_order = ?";
+
+    public GBFOrderGateKeeper(GBFOrderFinder orderFinder, GBFOrderTransmitter transmitter, String ddpInstanceName) {
+        this.orderFinder = orderFinder;
+        this.transmitter = transmitter;
+        this.ddpInstanceName = ddpInstanceName;
+    }
+
+
+    public void sendPendingOrders(Connection conn) {
+        for (SimpleKitOrder simpleKitOrder : orderFinder.findKitsToOrder(ddpInstanceName, conn)) {
+            Response orderResponse = transmitter.orderKit(simpleKitOrder.getRecipientAddress(), simpleKitOrder.getExternalKitName(), simpleKitOrder.getExternalKitOrderNumber(), simpleKitOrder.getParticipantGuid());
+
+            if (orderResponse.isSuccess()) {
+                try (PreparedStatement stmt = conn.prepareStatement(UPDATE_TRANSMISSION_DATE)) {
+                    //stmt.setTimestamp(1, Instant.now());
+                    stmt.setString(2, simpleKitOrder.getExternalKitOrderNumber());
+                    int numRowsUpdated = stmt.executeUpdate();
+
+                    // putting 10 things in a kit seems crazy?
+                    if (numRowsUpdated > 10) {
+                        throw new RuntimeException("Updated " + numRowsUpdated + " for order " + simpleKitOrder.getExternalKitOrderNumber() + ".  That seems like a lot?");
```

Comment:
I think I didn't understand this, if this is updated based on external order number, isn't 10 a bit too high?

Reply:
It's a bit of a guess--what's the maximum number of sub-items in a kit?  Or should we even check this at all?

Reply:
I see, so this is taking possible additions of subkits into account. Then yes I think we should check that, because as long as all of them are considered one kit, all of those rows in the table will get the same external order number.